packages/@aws-cdk/aws-lambda/lib/lambda-version.ts

```diff
@@ -223,6 +246,36 @@ export class Version extends QualifiedFunctionBase implements IVersion {
     return addAlias(this, this, aliasName, options);
   }
 
+  /**
+   * Enable autoscaling for the given lambda version. Requires that lambda version has provisioned concurrency.
+   * @param props The properties for autoscaling
+   */
```

Comment:
We may want to delete the capability of putting PC on a Version. 

The resource: AWS::Lambda::Version today is brittle and a sharp edge for many customers in how it behaves. We have a small flag [in our cfn docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-version.html#aws-resource-lambda-version-properties) which needs to be bolder, in red, and said twice for ProvisionedConcurrencyConfig. 

Versions do not support updates, so if you try to set a PC value on a version that already exists -- your cfn rolls back permanently. If you try to **only update** a PC configuration on a version cfn rolls back permanently. If you replace a version by changing the logical id, but set your version to retain-- your previous PC Configuration values are stranded, costing the customer. Overall, the Lambda team went back and forth whether we wanted to keep this functionality at all in cloudformation and decided "some customers may find a way to use it, and we should not inhibit them -- but direct everyone else to Alias-PC". 

I would like us to retain that same philosophy in the L2 by not exposing the sharp edge to customers --they can still use it in the L1 if they so desire.

Reply:
Sounds good, and thanks for the explanation. I'll remove `autoScaleProvisionedConcurrency()` from lambda-version.ts

Reply:
Also, should we update the application autoscaling [docs](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-applicationautoscaling#lambda-provisioned-concurrency-auto-scaling) where the `Lambda Provisioned Concurrency Auto Scaling` example autoscales on a lambda version? Or at least should we provide a disclaimer mentioning the pitfalls of autoscaling on a lambda version?